docroot/js/mediaplayer.js

```diff
@@ -4,380 +4,402 @@ var LMS = LMS || {};
+            _elems.progress.style.width =
+                "" +
+                ((_offset + _elems.audio.currentTime) / _duration) * 100 +
+                "%";
+            _elems.curtime.innerHTML = _durationToString(
+                _offset + _elems.audio.currentTime
+            );
+        });
+
+        _elems.audio.addEventListener("ended", function () {
+            Wt.emit(_root, "playbackEnded");
+        });
+
+        _elems.audio.addEventListener("canplay", function () {
+            if (_getAudioMode() == Mode.Transcode) {
+                _elems.transcodingActive.style.visibility = "visible";
+            } else {
+                _elems.transcodingActive.style.visibility = "hidden";
+            }
+        });
+
+        _initVolume();
+        _initDefaultSettings(defaultSettings);
+
+        _elems.volumeslider.addEventListener("input", function () {
+            _setVolume(_elems.volumeslider.value);
+        });
+
+        _elems.volume.addEventListener("click", function () {
+            if (_elems.audio.volume != 0) {
+                _setVolume(0);
+            } else {
+                _setVolume(_elems.lastvolume);
+            }
+        });
+
+        document.addEventListener("keydown", function (event) {
+            let handled = false;
+
+            if (event.target instanceof HTMLInputElement) return;
+
+            if (event.keyCode == 32) {
+                _playPause();
+                handled = true;
+            } else if (event.ctrlKey && event.keyCode == 37) {
+                _playPrevious();
+                handled = true;
+            } else if (event.ctrlKey && event.keyCode == 39) {
+                _playNext();
+                handled = true;
+            }
+
+            if (handled) event.preventDefault();
+        });
+
+        if ("mediaSession" in navigator) {
+            navigator.mediaSession.setActionHandler(
+                "previoustrack",
+                function () {
+                    _requestPreviousTrack();
+                }
+            );
+            navigator.mediaSession.setActionHandler("nexttrack", function () {
+                _requestNextTrack();
+            });
+        }
+    };
+
+    var _removeAudioSources = function () {
+        while (_elems.audio.lastElementChild) {
+            _elems.audio.removeChild(_elems.audio.lastElementChild);
+        }
+    };
+
+    var _addAudioSource = function (audioSrc) {
+        let source = document.createElement("source");
+        source.src = audioSrc;
+        _elems.audio.appendChild(source);
+    };
+
+    var _getAudioMode = function () {
+        if (_elems.audio.currentSrc) {
+            if (_elems.audio.currentSrc.includes("format"))
+                return Mode.Transcode;
+            else return Mode.File;
+        } else return undefined;
+    };
+    var loadTrack = function (params, autoplay) {
+        if (typeof window.audioUnlocked == "undefined") {
```

Comment:
Hi!
Thanks for this PR!
Does this mean that the track is not loaded if the user has not made any interaction yet?
What happens if a user (re)connects to LMS? Currently the last played track is loaded, but the track is not automatically played

(about the reformatting I planned to switch to clang-format)

Reply:
well you can't autoplay stuff, the problem  was generally 
```
 	var _audioCtx = new (window.AudioContext || window.webkitAudioContext)();
```
you will have to interact with the window before creating the context, that a rule in webkit based browsers, FF generally is a bit different having its own autoplay permission.

https://developer.chrome.com/blog/autoplay/#web-audio

I know the fix is not great, I just wanted to make as less changes as possible, about the formatting..sorry about that I just have a habbit to click reformat button all the time.